src/app/app.state.ts

```diff
@@ -4,8 +4,11 @@ import { authReducer } from './auth/auth.reducer';
 import { productReducer, ProductState } from './products/product.reducers';
 
 export interface User {
-  login_token: string | null;
-  refresh_token: string | null;
+  login_token: string | undefined;
```

Comment:
Why are all these stuffs possbily undefined. If user exist then he should have a token.